reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/common/ParseMarble.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2016 Facebook, Inc.
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ *  the License. You may obtain a copy of the License at
+ *  <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  <p>
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivesocket.tckdrivers.common;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.reactivesocket.Payload;
+import io.reactivesocket.util.PayloadImpl;
+import org.reactivestreams.Subscriber;
+
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * This class is rather straight forward for most interactions except for channel.
+ */
+public class ParseMarble {
```

Comment:
This can probably be turned into an `Iterable` that parses the file. Then you can just do this:

```java
Publisher p = RxReactiveStreams.toPublisher(Observable.from(marbleIterable));
```

Or if something more custom is needed:

```java
    Observable
        .<String> create(SyncOnSubscribe.<Iterator<String>, String> createStateful(
            () -> {
              // parse state ... (don't do this if just an Iterable ... use the above simpler example
              ArrayList<String> list = new ArrayList<String>();
              list.add("one");
              list.add("two");
              list.add("three");
              return list.iterator();
            } ,
            (iter, obs) -> {
              // state == position in iterable

             // this gets called for each onNext to be done so custom parsing could occur

              if (iter.hasNext()) {
                obs.onNext(iter.next());
              } else {
                obs.onCompleted();
              }
              return iter;
            }))
```